rails/app/controllers/dashboard/themes_controller.rb

```diff
@@ -11,6 +11,14 @@ def show
     def update
       @theme = current_community.theme
       if @theme.update(theme_params)
+        if current_community.public
+          static_map_url = if @theme.boundaries
+            "https://api.mapbox.com/styles/v1/mapbox/streets-v11/static/#{@theme.boundaries.flatten.to_s.gsub(' ','').sub("[", "%5B").sub("]", "%5D")}/900x600?access_token=#{@theme.mapbox_token}"
+          else
+            "https://api.mapbox.com/styles/v1/mapbox/streets-v11/static/#{@theme.center_long},#{@theme.center_lat},#{@theme.zoom},#{@theme.bearing},#{@theme.pitch}/900x600?access_token=#{@theme.mapbox_token}"
+          end
+          @theme.static_map.attach(io: URI.parse(static_map_url).open, filename: "#{current_community.slug}-static-map.png")
```

Comment:
I usually start factoring out once I hit three. I'm open to doing it now, but I also think it might be worth doing a more broad mapbox implementation audit and refactor since we have shared code in several spots now.